neuralpp/test/quick_tests/graphical_model/representation/belief_propagation_test.py

```diff
@@ -30,12 +31,56 @@ def prob_wet_grass(wetness: int, rain_level: int, sprinkler_on: int):
     ]
 
     expected_w = PyTorchTableFactor([w], [0.192, 0.332, 0.34, 0.136])
-    assert (BeliefPropagation().run(w, model) == expected_w)
+    assert (ExactBeliefPropagation(model, True).run(w) == expected_w)
+    assert (ExactBeliefPropagation(model, False).run(w) == expected_w)
 
     # observe cloudiness at highest level
     observations = {c: 2}
     conditioned_model = [f.condition(observations) for f in model]
 
     # this should result in increased chances of rain
     expected_w_with_conditions = PyTorchTableFactor([w], [0.12, 0.26, 0.42, 0.2])
-    assert(BeliefPropagation().run(w, conditioned_model) == expected_w_with_conditions)
+    assert(ExactBeliefPropagation(conditioned_model, True).run(w) == expected_w_with_conditions)
+    assert (ExactBeliefPropagation(conditioned_model, False).run(w) == expected_w_with_conditions)
+
+
+def test_ebp_with_loop():
```

Comment:
It would be good to keep the old tree-based test for Belief Propagation (just for completeness).

Reply:
Oh, sorry, I missed that.